flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfiguration.java

```diff
@@ -93,4 +97,82 @@ public Time getTtl() {
 	public TtlTimeCharacteristic getTimeCharacteristic() {
 		return timeCharacteristic;
 	}
+
+	@Override
+	public String toString() {
+		return "StateTtlConfiguration{" +
+			"ttlUpdateType=" + ttlUpdateType +
+			", stateVisibility=" + stateVisibility +
+			", timeCharacteristic=" + timeCharacteristic +
+			", ttl=" + ttl +
+			'}';
+	}
+
+	public static Builder newBuilder(Time ttl) {
+		return new Builder(ttl);
+	}
+
+	/**
+	 * Builder for the {@link StateTtlConfiguration}.
+	 */
+	public static class Builder {
+
+		private TtlUpdateType ttlUpdateType = OnCreateAndWrite;
+		private TtlStateVisibility stateVisibility = NeverReturnExpired;
+		private TtlTimeCharacteristic timeCharacteristic = ProcessingTime;
+		private Time ttl;
+
+		public Builder(Time ttl) {
```

Comment:
Should `TimeCharacteristic` be a builder's constructor param as well?  Otherwise, users may not notice they have to set it, and may easily lead into confusion

Reply:
Actually, I do not see a problem to have processing time as a default value for `TimeCharacteristic`. I think event time is more specific and users who need it will deliberately look for it. First release of TTL feature will even have only processing time option atm.

Reply:
@azagrebin I removed the default value for TimeCharacteristic, it will depend on users' choice. Do you agree this mode?

Reply:
I suggest to revert where it processing time was default value as it is the only option atm and keep only ttl as a mandatory parameter.